tests/Aspire.Hosting.Tests/Utils/TestDistributedApplicationBuilder.cs

```diff
@@ -52,6 +54,9 @@ private static IDistributedApplicationTestingBuilder CreateCore(string[] args, A
     {
         var builder = DistributedApplicationTestingBuilder.Create(args, (applicationOptions, hostBuilderOptions) => configureOptions?.Invoke(applicationOptions));
 
+        // Load the user secrets file for Aspire.Hosting.Tests
+        builder.Configuration.AddUserSecrets<AspireStoreTests>();
```

Comment:
Why is this necessary? I don't see any reasoning in the comments or in the PR description.

Reply:
If this really is necessary, it shouldn't reference `AspireStoreTests`. But instead:

```suggestion
        builder.Configuration.AddUserSecrets<TestDistributedApplicationBuilder>();
```

Reply:
> The reason is that passwords need to be reused across restarts for persistent containers snapshots to be deterministic. 

We do that in other tests by getting the password from the first resource and flowing it into the 2nd resource.

https://github.com/dotnet/aspire/blob/c36342ccdf439830b26389379454fb9f2184bae8/tests/Aspire.Hosting.PostgreSQL.Tests/PostgresFunctionalTests.cs#L205-L209

I don't want the tests mucking with user secrets on the file system. Especially in my User directory. There could be issues with persisting this information in a non-TEMP folder. We want the tests to be "clean" and not mutate the machine state.

Reply:
Ok, but it's fine only because the store is unit tested in other places and we'll assume it works fine so this test doesn't have to use this component. It does make it less of a functional test though.

Do the playground tests follow the same principle though?

Reply:
Done. One thing I fixed though is that the aspire store path needs to be constant between runs, since it's defined randomly with each builder.